### PR TITLE
Short Codes SDK as Private Preview

### PR DESCRIPTION
**Pull request by guilhermeluizsp**
# Short Codes SDK

**DISCLAIMER: SHORT CODES SDK IS ONLY FOR INTERNAL USERS. THIS SDK WILL NOT BE PUBLICLY SUPPORTED.**

This PR introduces a new SDK for managing Short Codes. Short Codes are a type of number that are available to enterprise customers. They come in the form of a 5 or 6 digit number and can be used to send sms. [We recently released them for public preview](https://docs.microsoft.com/en-us/azure/communication-services/quickstarts/sms/apply-for-short-code). **This is intended to be released as private preview**, as it will only be used internally.

## Important notes

- The swagger file was copied from the [spec repo](https://github.com/Azure/azure-rest-api-specs/tree/main/specification/communication/data-plane/ShortCodes) but passed through some modifications, as detailed in the autorest.md file.

### CODE REVIEW

File: sdk/communication/ci.yml [hunk 1]
```diff
@@ -26,6 +26,8 @@ extends:
     ServiceDirectory: communication
     ArtifactName: packages
     Artifacts:
+    - name: Azure.Communication.ShortCodes
```
Comment (chidozieononiwu) [inline sdk/communication/ci.yml:29]: Adding this here might may lead to accidentally publishing the package to the public nuget feed. Unless there is not internal pipeline created for this

Comment (guilhermeluizsp) [reply sdk/communication/ci.yml:29]: @chidozieononiwu But isn't it also necessary for releasing as private preview?

Comment (guilhermeluizsp) [reply sdk/communication/ci.yml]: I see. I just removed it, thanks!

**Review by chidozieononiwu — APPROVED** [review]

**Pull request merged**